src/ModelFactory.php

```diff
@@ -54,7 +54,14 @@ public function __construct()
      */
     final public static function findOrCreate(array $attributes): Proxy
     {
-        if ($found = static::repository()->find($attributes)) {
+        // remove attribute prefixes (like "force:" and "optional:")
+        $normalizedAttributes = [];
+        foreach ($attributes as $key => $value) {
+            $normalizedKey = false !== \mb_strpos($key, ':') ? \mb_substr($key, \mb_strpos($key, ':') + 1) : $key;
+            $normalizedAttributes[$normalizedKey] = $value;
```

Comment:
Also, `optional:` attributes be removed from the array, no?

Reply:
Oh, good catch. Included that one as well